examples/demo-rollup/src/ledger_rpc.rs

```diff
@@ -19,39 +22,48 @@ use self::query_args::{extract_query_args, QueryArgs};
 ///    Example Query: `curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"ledger_getBatches","params":[[1, 2], "Full"],"id":1}' http://127.0.0.1:12345`
 /// - ledger_getEvents
 ///    Example Query: `curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"ledger_getBatches","params":[1, 2],"id":1}' http://127.0.0.1:12345`
-fn register_ledger_rpc_methods<B: Serialize + DeserializeOwned, T: Serialize + DeserializeOwned>(
+fn register_ledger_rpc_methods<
+    B: Serialize + DeserializeOwned + Clone + 'static,
+    T: Serialize + DeserializeOwned + Clone + 'static,
+>(
     rpc: &mut RpcModule<LedgerDB>,
 ) -> Result<(), jsonrpsee::core::Error> {
     rpc.register_method("ledger_getHead", move |_, db| {
-        db.get_head::<B, T>().map_err(|e| e.into())
+        db.get_head::<B, T>()
+            .map_err(|e| to_jsonrpsee_error_object(e, LEDGER_RPC_ERROR))
     })?;
 
     rpc.register_method("ledger_getSlots", move |params, db| {
-        let args: QueryArgs<SlotIdentifier> = extract_query_args(params)?;
-        db.get_slots::<B, T>(&args.0, args.1).map_err(|e| e.into())
+        let args: QueryArgs<SlotIdentifier> = extract_query_args(params).unwrap();
+        db.get_slots::<B, T>(&args.0, args.1)
+            .map_err(|e| to_jsonrpsee_error_object(e, LEDGER_RPC_ERROR))
     })?;
 
     rpc.register_method("ledger_getBatches", move |params, db| {
-        let args: QueryArgs<BatchIdentifier> = extract_query_args(params)?;
+        let args: QueryArgs<BatchIdentifier> = extract_query_args(params).unwrap();
```

Comment:
Under what situations can these `.unwrap()` panic? And does the whole server crash if they do?

Reply:
Fixed